### ISSUE TITLE
Windows Narrator doesn't say selection state for SwatchColorPicker on keyboard selection.

### ISSUE THREAD

**Issue by davidprejban** [OfficeDev/office-ui-fabric-react#5424]
### Bug Report 
- __Package version(s)__: 
6.21.0

- __Browser and OS versions__:
Windows server 2016: Version 1607 OS Build 1493.2339
Internet Explorer 11: Version 11.2239.14393.0
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix?
Yes
 
Requested priority:
High
 
#### Describe the issue:
Windows narrator will not say whether or not the color is selected when selecting the color.
 
Reproduction steps:
1. Open I.E 11
2. Navigate to:
  a. https://developer.microsoft.com/en-us/fabric#/components/checkbox
3. Turn on windows narrator
  a. https://support.microsoft.com/en-us/help/22817
4. Tab until you have a color in the swatch color picker example is selected
5. Use arrows to move to unselected color
6. Wait for narration to end
7. Press 'space' or 'enter' on the keyboard to select color

#### Actual behavior:
Narrator says the key that was pressed. 
E.g. "space" if 'space' key is pressed.
 
#### Expected behavior:
Narrator says that the color is selected. 
E.g. "blue selected" if 'space' or 'enter' key is pressed.

**Comment by philipzloh** [comment]
I have bigger issues with the SwatchColorPicker... and that's even with its visual representation.

5424, as an issue, though, is either with Narrator or runs deeper. The role used for each color here is a `gridcell`, which only supports `aria-selected` because you imagine selecting/multiselecting a grid or multiple grids like in Excel, not a `button` which has state or a `radio` or a `checkbox`. Therefore the SwatchColorPicker needs to be rewritten heavily in order to be accessible at all.

**Comment by dzearing** [comment]
@jspurlin Just a heads up here; I believe you wrote the component originally. Maybe we can talk about what's wrong, and plan for an updated version.

**Comment by dzearing** [comment]
I'm having a difficult time understanding what the outcome should be here. It seems that there is no 1 single fix here to remedy this bug, and that @philipzloh is saying the whole thing needs to be redone. It would be excellent to be crisp about the outcome and not say "this one thing is broken, but even if its fixed, doesn't matter because there are way more issues than this..."

What are the things that need to be fixed? @philipzloh can you list what you'd like to see here? From there, we can assess when it can be rewritten, or if the issues can be individually addressed.
